bindings/tflite/java/org/tensorflow/lite/Interpreter.java

```diff
@@ -0,0 +1,430 @@
+ * }</pre>
+ *
+ * <p>If a model takes multiple inputs, use
+ *   {@link #runForMultipleInputsOutputs(Buffer[], Map)}:
+ *
+ * <pre>{@code
+ *  // Load model/initialize interpreter same as above.
+ *
+ *  // Prepare inputs:
+ *  Buffer[] inputs = {input0, input1, ...};
+ *  Map<Integer, Buffer> indexToOutput = new HashMap<>();
+ *  FloatBuffer ithOutput = ... allocate (native) and populate buffer ...
+ *  indexToOutput.put(i, ithOutput);
+ *
+ *  // Run inference:
+ *  interpreter.runForMultipleInputsOutputs(inputs, indexToOutput);
+ *
+ *  // Cleanup same as above.
+ * }</pre>
+ *
+ * <p>Orders of inputs and outputs are determined when converting TensorFlow
+ * model to TensorFlow Lite model with TOCO, as are the default shapes of the
+ * inputs.
+ *
+ * <p><b>WARNING:</b>Instances of {@link Interpreter} are <b>not</b> thread-safe.
+ * A {@link Interpreter} owns resources that <b>must</b> be explicitly freed by
+ * invoking {@link #close()}
+ */
+public final class Interpreter implements AutoCloseable {
+  private static final String TAG = Interpreter.class.getCanonicalName();
+
+  /**
+   * Options class for controlling runtime interpreter behavior.
+   *
+   * <p>Currently only {@link #setNumThreads(int)} is supported.
+   */
+  public static class Options {
+    int numThreads = -1;
+
+    public Options() {}
+
+    /**
+     * Sets the number of threads to be used for ops that support multi-threading.
+     *
+     * <p>{@code numThreads} should be >= -1. Setting {@code numThreads} to 0 has the effect to
+     * disable multithreading, which is equivalent to setting {@code numThreads} to 1. If
+     * unspecified, or set to the value -1, the number of threads used will be
+     * implementation-defined and platform-dependent.
+     */
+    public Options setNumThreads(int numThreads) {
+      this.numThreads = numThreads;
+      return this;
+    }
+  }
+
+  private final int inputTensorCount;
+  private final int outputTensorCount;
+  private final Tensor[] inputTensors;
+  private final Tensor[] outputTensors;
+  private final long nativeAddress;
+
+  private long inferenceDurationNanoseconds;
+  private boolean tensorsAllocated;
+
+  /**
+   * Initializes the Interpreter with model and options.
+   *
+   * @param modelByteBuffer: a directly allocated, native {@link java.nio.ByteOrder} byte buffer of
+   *     an IREE compatible TFLite model.
+   * @param options: options for the interpreter, or null (to use defaults).
+   * @throws IllegalArgumentException if the model cannot be initialized in the Interpreter.
+   */
+  public Interpreter(@NonNull ByteBuffer modelByteBuffer, Options options)
+      throws IllegalArgumentException {
+    TensorFlowLite.init();
+    if (options == null) {
+      options = new Options();
+      options.setNumThreads(2);
+    }
+    nativeAddress = nativeNew(modelByteBuffer, options.numThreads);
+
+    if (nativeAddress == 0) {
+      throw new IllegalArgumentException("Could not create Interpreter");
```

Comment:
since you are throwing here there's never a case where `nativeAddress` can be 0 on an `Interpreter` and thus you don't need any of the `checkInitialized` calls